plutus-core/plutus-ir/src/PlutusIR/Analysis/Dependencies.hs

```diff
@@ -60,42 +66,27 @@ varStrictnessFun = do
 runTermDeps
     :: (DepGraph g, PLC.HasUnique tyname PLC.TypeUnique, PLC.HasUnique name PLC.TermUnique,
        PLC.ToBuiltinMeaning uni fun)
-    => Term tyname name uni fun a
+    => PLC.BuiltinVersion fun
+    -> Term tyname name uni fun a
     -> (g, StrictnessMap)
-runTermDeps t = flip runState mempty $ flip runReaderT Root $ termDeps t
-
--- | Compute the dependency graph of a 'Type'. The 'Root' node will correspond to the type itself.
---
--- This graph will always be simply a star from 'Root', since types have no internal let-bindings.
---
--- For example, the graph of @(all a (type) [f a])@ is:
--- @
---     ROOT -> f
---     ROOT -> a
--- @
---
-runTypeDeps
```

Comment:
You are curious why I removed this code. First it's dead code so that is fine.
Second is that typedeps *do not need need the version* in the context, but they happen to share the same reader DepCtx with term deps.
Instead of going the (Maybe ver) route, I decided to just remove this code and leave the ctx not specified.

@michaelpj @effectfully that is what i was trying to avoid with my ugly monadtrans approach. here i am duplicating context which is actually not shared at all.

Reply:
> If you are curious why I removed this code. First it's dead code so that is fine.

I think this reason is enough of justification for the removal.